docs/installing-solidity.rst

```diff
@@ -527,23 +527,12 @@ If you are interested what CMake options are available run ``cmake .. -LH``.
 
 SMT Solvers
 -----------
-Solidity can be built against Z3 SMT solver and will do so by default if
-it is found in the system. Z3 can be disabled by a ``cmake`` option.
-
-*Note: In some cases, this can also be a potential workaround for build failures.*
-
-
-Inside the build folder you can disable Z3, since it is enabled by default:
-
-.. code-block:: bash
-
-    # disables Z3 SMT Solver.
-    cmake .. -DUSE_Z3=OFF
+Solidity can optionally use SMT solvers, namely ``z3``, ``cvc5`` and ``Eldarica``,
+but their presence is checked only at runtime, they are not needed for the build to succeed.
 
 .. note::
 
-    Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,
-    but their presence is checked only at runtime, they are not needed for the build to succeed.
+    The emscripten builds require Z3 and will statically link against it instead.
```

Comment:
It actually *is* a hard requirement now. I'm actually wondering if we should make it optional for the emscripten builds, but if we want to do that, we can do it separately (and custom emscripten builds ought to be *very* rare anyways)